python/ray/experimental/serve/api.py

```diff
@@ -67,27 +67,78 @@ def create_backend(func_or_class, backend_tag, *actor_init_args):
             initialization method.
     """
     if inspect.isfunction(func_or_class):
-        runner = TaskRunnerActor.remote(func_or_class)
+        # ignore lint on lambda expression
+        creator = lambda: TaskRunnerActor.remote(func_or_class)  # noqa: E731
     elif inspect.isclass(func_or_class):
         # Python inheritance order is right-to-left. We put RayServeMixin
         # on the left to make sure its methods are not overriden.
         @ray.remote
         class CustomActor(RayServeMixin, func_or_class):
             pass
 
-        runner = CustomActor.remote(*actor_init_args)
+        # ignore lint on lambda expression
+        creator = lambda: CustomActor.remote(*actor_init_args)  # noqa: E731
     else:
         raise TypeError(
             "Backend must be a function or class, it is {}.".format(
                 type(func_or_class)))
 
-    global_state.backend_actor_handles.append(runner)
+    global_state.backend_creators[backend_tag] = creator
 
-    runner._ray_serve_setup.remote(backend_tag,
-                                   global_state.router_actor_handle)
+    global_state.registered_backends.add(backend_tag)
+
+    scale(backend_tag, 1)
+
+
+def _start_replica(backend_tag):
+    assert backend_tag in global_state.registered_backends, (
```

Comment:
Going forward, it would be good to have well-defined exceptions for everything, and make it part of the method documentation, instead of just having assertion errors :)